crates/xilem_html/web_examples/todomvc/src/state.rs

```diff
@@ -48,21 +78,42 @@ impl AppState {
             self.editing_id = Some(id)
         }
     }
+
+    /// Load the current state from local_storage, or use the default.
+    pub fn load() -> Self {
+        let Some(raw) = storage().get_item(KEY).unwrap_throw() else {
+            return Default::default();
+        };
+        match serde_json::from_str(&raw) {
+            Ok(todos) => todos,
+            Err(e) => {
+                tracing::error!("couldn't load existing todos: {e}");
+                Default::default()
+            }
+        }
+    }
+
+    /// Save the current state to local_storage
+    pub fn save(&self) {
+        let raw = serde_json::to_string(self).unwrap_throw();
```

Comment:
I'm not sure if `serde_json` isn't a little bit too heavy for this. I'm thinking about something like `serde_wasm_bindgen`, but I think we can optimize that later.